python/paddle/fluid/framework.py

```diff
@@ -913,6 +913,15 @@ def sync_with_cpp(self):
                     ops_in_cpp_index += 1
                     ops_in_python_index += 1
 
+        # sync ops inserted from c++ end
+        if len(self.ops) != len(ops_in_cpp) and start_index == 0 and len(
+                self.ops) == end_index:
+            self.ops.clear()
+            for index in range(len(ops_in_cpp)):
```

Comment:
For simplicity, can we only keep this and remove all above sync operations? It's slower but very easy to understand.

Reply:
Sorry, I couldn't remove all above sync operators. I tried it, but many unit tests fail. The reason I guess is:
```
# sync ops append to the head of cpp_ops
for index in range((start_index - 1 - 1), -1, -1):
    op_desc = ops_in_cpp[index]
    op = Operator(self, op_desc)
     self.ops.appendleft(op)
```
The reverse order of `range((start_index - 1 - 1), -1, -1)` and `appendleft` make the order of `python_ops` different from `cpp_ops`.

However, `sync ops append to the end of cpp_ops`, `sync ops removed from c++ end`, `sync ops inserted from c++ end` make the order of `python_ops` the same as `cpp_ops`.  

Thus, can we only remain the codes of `sync ops append to the head of cpp_ops`, and remove others?

Reply:
That's wired. The above code still tries to make the order of `self.ops` the same with `cpp_ops`. We can merge this if it's currently a blocking feature and add some comments here.